examples/sampling-monomial-gamma-hmc.ipynb

```diff
@@ -0,0 +1,413 @@
VhW6G8JURpvwYjjqSR\ngaYbkamtZYaJRJe2r1BJnBCeZH4P8cLkVb0AkrLIY6l2MrbuhwiGQqhwPWFlYwgJUnjGlxN2Vm4W\nhAw5NN+IdX02SNsKNSv2cQ2sE/o8kt7ca2WqW7kHK3D+ipLwIhoKH/S2FgyL0eEkl9Xb2FpYVudF\nyYiN2gXfwLJkWZMvl1Etyyh5pB+y5sicEV5/J8dzNMcsdUjrGk3TjFOF0CNnWCyQkogozVdEGnSO\nZLjEr+jnFhcojnJR+t0KWTTpKMlegQjDN8CjLMp3IefWhuYC8PugxdOOEj5PhGvlSaSfB2dIx1Nz\nsXYRRUoh9jnDdr0v0jdUvIUJs+y9hwAiegYpBZaQ1KLRqpWabge9ZEHjP5YtogXuvJfj4303WlJP\nc83cz1C0M1ySO4YmgImloXPHszqHMikuNXS8+1Vu9FvY6KK6MeJfi1vUIqoJTFu63lovNM6VndJ8\ne73wPCKBZ2o+c8lTeh4028npdJYZRMhwigZDmwOhq5KSp1q64Y+izMNr5xOI5EDoMxOb32gnaHRN\nlMn+5pWbUpZ0T1TFu+uNcdI+yDPIdMu0bxNCfAi4Ukr5mhCiAabOs1UoFArF6bNz5042b95MS0sL\ndXV1jI+Po+vVlarzid///d/njjvuQAhBa2srf/7nf057e/vJDzx7/F9CiC8Dl0kp/6cQYjmwTEo5\ndUOvWSConBdtKxQdVbQqjaZqr6N2kbrMPlIUqyomNpJeUm7eRml7Vhac7CbbZEzkWCJL4XamqKPW\n6Z5FvP4SipFaJBqaNIjaBUytHoFkzM4xQSEkUMbOQKSBBfleGswJBud9xJfDl11CxvWGBQ3A4LVZ\ndulq3uwbwcBmIfXE3fMJbNeoKR0ppAzPgQgoZQJsImBboVLd7iaQFraU5Myg0RO4IZLKSoOBsC67\nzAMUIYoR8IB4YlnSkb1o2NiBtl2e10dKo2qlPrOKgSXB0epMKEzW8FWGGz8LaSGR5DCYY0tKlRwg\nr1s01pWUXM3vzuNkpHnXv2JiB6W2yiU0TAq2xY6uOPk6i0wmz0oCxkPZ4ymxMSzJE7t7sdyx66xw\naKQjs3PgmH4UgAZpE/b3ukaKtJ0iFZQUbG8NIxKYVC+oL1pMcL1cyn4xwpiozFX29w8+UjIcHmm5\nGxfke0Euod5M0ZV8BqLzSpIV0xXXnqRIupgvyV72TBaFxTHhFdeo9EwTMH1qa2opGM48lRtqjpEj\n/COCWzVp+XObkBnqcL53NhDFJGeYrjc5vHARCTwzZsDA9Pb1JE6jV3jfao+/CplwesYut+Wu15Lg\noD1OOfNlHROBexgjT6sxQkTUu8Zw2D1a+ts6lX9rZpmsiXYIIcTdwHPAw+5HFwMvnS2hFArF+UVL\n6mlaUk/zS+1Y1X+KmWdoaIjm5mbq3BiU5uZmLrroIgDuvfdebr75Zq6//nruuecef7V87dq1fOtb\n3+JTn/oU11xzDXv37uVzn/scV155Jd/97ncBp9Hw1VdfzVe+8hVuuOEGPv/5z5PL5SrO/+qrr/Lx\nj3+cj33sY3zhC18gk3FCjL7zne9w7bXXcsMNN/Dtb3+74rjGxkY/vCSbzVaEmswCPwI+DvyF+z7j\nfnbe4Snwlpy8nPCkq7nuMxCRBuPpV3lD66u6Z1wUOKCN+h6NIIvlHIS0OC4SvKwd54Bwwn1k4B6a\nWgQpbaRwEtYvzBziktR+lqUPYtuSHVp4Ndq7pkZjLOAFCKt26aLpK6aTOegsWTLKfm11kxSOAXm7\nvQJwlW5ZntkVDkCSaKHyehdFloNtkaNyLvZ0D/JW1xhHx0rfjWCoW043A4aKp6KXrqrMviIiom6x\niUqEdCqyWQHN3fP6BL2Ji/JdfmifKauXwy4KZ46HZTYgGSyYU0ON5ijWlpRYmuM90iONjNlDvKb1\nME4+NH/lr2utHCI0x44CHJlEFk1aGLaNgcZ4tmQQzjEn3DErb/ZLB5znxzMw5hnhQiLS3aabE8xP\n72euPs7ifG/ZKKX7oUmLcXIVRnREBg0sp7iKBL8U+lSEDNSycb371WDGseMnaM4dpTl3tLRdOtUl\ng1euY7NN6+PX3S/5BmB5FcFOUapcOF+EC27s04YxZCGweBBxX1dfYPH+Vpgi3Ih7Ub6bOjPcGDrj\n9rnTbCc/MF10ZA+OHIgQpC/pfV+cu+sVo5FSkhMm9UYyNGfVqvrFhWNUewVfiqaNboWvZSH1ob9L\nA8KtEGkWWZw/UTGm78E6h0UupmVgAX8D3AqOv1BKeRRYOuURgBCiWwhxUAhxQAixz/1skRDit0KI\no+7/C082jkJxNth5Ihb615fIh/4pFLPJ7bffTl9fH1dddRXr1q1j27Zt/rZvfOMb7N27l7a2NvL5\nPJs3b/a31dbWsn37dtG+FIQAACAASURBVL7+9a9z55138qMf/Yi2tjY2btxILObk0HV0dHDPPffQ\n2tpKU1MTGzaEwx/Hx8dZv349r732Gi0tLaxevZof/vCHxONxXnzxRQ4dOkRra6tvtJXz4osvcvXV\nV/OHf/iH/OxnPzsLs3NKrJFS/g04rg4p5QRQe7KDhBDLhRBbhRBHhBCHhBD/5WwKKaUMKeXZaFiJ\nusm+gGY5J6wySfhIXTPg5CEFaayN0FhXGWgSLCPurVZH7SL1VopL5TyEXTI2+oRXlKKkKpiixjWw\nyscOZwWV+hSF5ZpjTLCw0FtR5KJc/nLGMsWKlfMMBlpoxV/6p5NU8WAR9EIJ5mrz/ZX1IEVpsan3\n33lXjIX7HwXmIW/aJPIG41kd05e9FJ5kS8lFstHfP0KkVKjCDXcsRuY5pd+xQxXknJGcc6XKfDML\nCo6vqLewr+o8eYaE4T5MGja10pmlhQ21gMS2JbZ/LYKY6yHYLQZJBM5X3vOowYizKN+N71KTFtJK\nM246xVhWyvkhWYR0
```

Comment:
Something funny going on here. There's a tiny histogram inside the larger histograms

Reply:
I confess I don't see this on the rendered output? Looks ok to me.

Reply:
![Screenshot from 2019-09-11 16-30-14](https://user-images.githubusercontent.com/517644/64711790-1b5f2c00-d4b2-11e9-97c6-0c9e110ea53a.png)

Should disappear from just re-running the notebook I guess?

Reply:
![histo-in-histo](https://user-images.githubusercontent.com/517644/64711851-2dd96580-d4b2-11e9-8d75-5ff8703bff71.png)

Should disappear from just re-running the notebook I guess?

Reply:
Looks like an issue with the trace function as it's still there.

Reply:
Very strange!
I'll make an issue for this then

Reply:
Wait, before I do that @ben18785 , can you check if there's a plt.figure() or a plt.show() call between now and the last time the trace function was called? Maybe it's updating an existing graph!